README.md

```diff
@@ -1,37 +1,78 @@
 
-As usual please start by
 
-* Forking this repo
 
-* Finally submit a pull request before Monday week at 9am with your solution or partial solution.  However much or little amount of code you wrote please please please submit a pull request before Monday week at 9am. 
+## Planning - handling the rules
 
-___STRONG HINT, IGNORE AT YOUR PERIL:___ Bowling is a deceptively complex game. Careful thought and thorough diagramming — both before and throughout — will save you literal hours of your life.
 
-## Focus for this challenge
-The focus for this challenge is to write high-quality code.
 
-In order to do this, you may pay particular attention to the following:
-* Using diagramming to plan your approach to the challenge
-* TDD your code
-* Focus on testing behaviour rather than state
-* Commit often, with good commit messages
-* Single Responsibility Principle and encapsulation
-* Clear and readable code
+
+                                     ┌───────────────┐
+                                     │               │
+                                     │ SCORES ARRAY  │
+                                     │               │
+                                     └───────┬───────┘
+                                             │
+                                             ▼
+
+                                     Sum of each frame
+
+                                             │
+                                             │
+                                             │
+                                             ▼
+
+                          spare if frame == 10 && frame.length == 2
+                 ┌───────
+                 │        strike if frame == 10 && frame.length == 1
+                 │
+                 │                              │   │
+                 │                              │   │
+                 │                              │   │
+                 │              SPARE    ◄──────┘   └───────────────►    STRIKE
+                 │
+                 │    if spare, add the first roll               if strike, add the first roll of the
+                 │    of the next frame onto score               next frame onto the score
+                 │
+                 │                                                       │             │
+                 ▼                                                       │             │
+                                                          ┌──────────────┘             └────────────┐
+              10TH FRAME                                  ▼                                         ▼
+
+      3rd roll if first 2 rolls include      if strike, add the first roll            if not strike, add 2nd
+      strike(s) or spare                     next frame on                            roll of next frame on
+
+
+
+## I/O examples
```

Comment:
I really like the use of an input and output table. It explains what the logic of the class is and what to expect.